Detectors/TPC/base/include/TPCBase/EfficiencyGEM.h

```diff
@@ -0,0 +1,204 @@
+
+namespace o2 {
+namespace TPC {
+
+/// \class EfficiencyGEM
+
+class EfficiencyGEM{
+  public:
+    EfficiencyGEM();
+    ~EfficiencyGEM() = default;
+   
+    void setGeometry(int geom);  //Set GEM geometry (1 standard, 2 medium, 3 large)
+    
+    float getCollectionEfficiency(float ElecFieldRatioAbove);
+    float getExtractionEfficiency(float ElecFieldRatioBelow);
+    
+    float getGeometryThickness();
+    
+  private:    
+    
+    //Geometric parameters for the GEM efficiencies
+    float getParameterC1();
+    float getParameterC2();
+    float getParameterC3();
+    float getParameterC4();
+    float getParameterC5();
+    float getParameterC6();
+    float getParameterC7(float eta1, float eta2); 
+    float getParameterC8(float eta1, float eta2); 
+    float getParameterC9(float eta1, float eta2); 
+    
+    float getParameterC7Bar(float eta1, float eta2);  
+    float getParameterC8Bar(float eta1, float eta2);  
+    float getParameterC9Bar(float eta1, float eta2);  
+    
+    float getParameterC7BarFromX(float IntXStart, float IntXEnd);
+    float getParameterC8BarFromX(float IntXStart, float IntXEnd);
+    float getParameterC9BarFromX(float IntXStart, float IntXEnd);
+    
+    //Cathode flux calculations
+    float getLambdaCathode();
+    float getLambdaCathodeF2(int n);
+    float getLambdaCathodef2();
+    
+    float getMu1Cathode();
+    float getMu1CathodeF2(int n);
+    float getMu1Cathodef2();
+    
+    float getMu2Cathode();
+    float getMu2CathodeF2(int n);
+    float getMu2Cathodef2();
+    
+    
+    //Top electrode flux calculations
+    float getMu2Top(float IntXStart, float IntXEnd);
+    
+    float getMu2TopF2(int n, float IntXStart, float IntXEnd);
+    float getMu2TopFTaylorTerm0(int n);
+    float getMu2TopFTaylorTerm2(int n);
+    
+    float getMu2Topf2(float IntXStart, float IntXEnd);
+    float getMu2TopfTaylorTerm0();
+    float getMu2TopfTaylorTerm2();
+    
+    //Kink calculations
+    float getIntXEndBot(float eta1, float eta2);
+    float getIntXEndTop(float eta1, float eta2);
+    float getEta1Kink1(float eta2);
+    float getEta1Kink2(float eta2);
+    float getEta2Kink1(double eta1);
+    float getEta2Kink2(double eta1);
+    float getHtop0();
+    float getHtop2();
+    float FlipDistanceNextPrevStage();
+    
+    //GEM geometry definitions as it was used in fits
+    int mNumberHoles;
+    float mGeometryThickness;
+    float mGeometryPitch;
+    float mGeometryHoleDiameter;
+    float mGeometryWidth;
+    float mGeometryDistancePrevStage;
+    float mGeometryDistanceNextStage;
+    
+    //Field configuration as it was used in fits
+    float mElecFieldAbove;
+    float mElecFieldBelow;
+    float mPotentialGEM;
+    float mElecFieldGEM;
+    
+    //Scaling parameters from fits
+    float mGeometryTuneEta1;
+    float mGeometryTuneEta2;
+    float mGeometryTuneDiffusion;
+    
+    static constexpr float sPi = 3.1415926;
+    static constexpr float sKappa = 0.514;
```

Comment:
Very nice documentation in the beginning. Please also add short comments to all functions and member variables.